File: helfomat-api/src/main/java/de/helfenkannjeder/helfomat/api/organisation/event/OrganizationEditAddGroupEventDto.java
package de.helfenkannjeder.helfomat.api.organisation.event;

import de.helfenkannjeder.helfomat.api.organisation.GroupDto;
import de.helfenkannjeder.helfomat.core.organisation.OrganisationId;

/**
 * @author Valentin Zickner
 */
public class OrganizationEditAddGroupEventDto extends OrganizationEventDto {
    private int index;
    private GroupDto group;

    OrganizationEditAddGroupEventDto() {
    }

    public OrganizationEditAddGroupEventDto(OrganisationId organisationId, int index, GroupDto group) {
        super(organisationId);
        this.index = index;
        this.group = group;
    }

    public int getIndex() {
        return index;
    }

    public GroupDto getGroup() {
        return group;
    }
}

File: helfomat-api/src/main/kotlin/de/helfenkannjeder/helfomat/api/organization/event/OrganizationEditAddGroupEventDto.kt
package de.helfenkannjeder.helfomat.api.organization.event

import de.helfenkannjeder.helfomat.api.organization.GroupDto
import de.helfenkannjeder.helfomat.core.organization.OrganizationId

/**
 * @author Valentin Zickner
 */
data class OrganizationEditAddGroupEventDto(
    override val organizationId: OrganizationId,
    val index: Int,
    val group: GroupDto,
    override val eventApplicable: Boolean
) : OrganizationEventDto {

    override fun <T> visit(visitor: OrganizationEventDtoVisitor<T>): T {
        return visitor.visit(this)
    }

}